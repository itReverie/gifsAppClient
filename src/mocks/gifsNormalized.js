const gifsNormalized={ "B0vFTrb0ZGDf2": { id: "B0vFTrb0ZGDf2",
                                          isFavorite:true,
                                          preview :{
                                                url: "https://media1.giphy.com/media/B0vFTrb0ZGDf2/200w.gif?cid=4647a4e05b37e8f8747833465127ae26",
                                                width: "200",
                                                height: "105",
                                                size: "74046",
                                                mp4: "https://media1.giphy.com/media/B0vFTrb0ZGDf2/200w.mp4?cid=4647a4e05b37e8f8747833465127ae26",
                                                mp4_size: "28128",
                                                webp: "https://media1.giphy.com/media/B0vFTrb0ZGDf2/200w.webp?cid=4647a4e05b37e8f8747833465127ae26",
                                                webp_size: "146398"}
                                  },
                     "oDLDbBgf0dkis" : { id: "oDLDbBgf0dkis",
                                        isFavorite:true,
                                        preview: {
                                                    url: "https://media3.giphy.com/media/oDLDbBgf0dkis/200w.gif?cid=4647a4e05b37e8f8747833465127ae26",
                                                    width: "200",
                                                    height: "113",
                                                    size: "534324",
                                                    mp4: "https://media3.giphy.com/media/oDLDbBgf0dkis/200w.mp4?cid=4647a4e05b37e8f8747833465127ae26",
                                                    mp4_size: "85697",
                                                    webp: "https://media3.giphy.com/media/oDLDbBgf0dkis/200w.webp?cid=4647a4e05b37e8f8747833465127ae26",
                                                    webp_size: "53300"}
                                }   
};

export default gifsNormalized;

