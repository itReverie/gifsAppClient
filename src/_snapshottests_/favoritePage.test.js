import React from 'react';
import FavoritePage from '../pages/favoritePage';
import renderer from 'react-test-renderer';
import gifsNormalized from '../mocks/gifsNormalized';
import {BrowserRouter} from 'react-router-dom';


it('rendering correctly a list of favorite gifs page via snapshot', () => {
  const onAddFavorite=jest.fn();

  const tree = renderer
    .create(<BrowserRouter><FavoritePage favoriteGifs={gifsNormalized}
                                         onAddFavorite={onAddFavorite} /></BrowserRouter>)
    .toJSON();
  expect(tree).toMatchSnapshot();
});