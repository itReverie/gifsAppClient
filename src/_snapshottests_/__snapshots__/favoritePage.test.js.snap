// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`rendering correctly a list of favorite gifs page via snapshot 1`] = `
<div
  className="sc-gZMcBi ktHDkb"
>
  <div
    className="sc-bwzfXH hDAGxY"
  >
    <a
      href="/"
      onClick={[Function]}
    >
      <img
        className="sc-htpNat gCTdIZ"
        src="home.svg"
      />
    </a>
    <a
      href="/"
      onClick={[Function]}
    >
      <img
        className="sc-bxivhb ffNsIf"
        src="back.svg"
      />
    </a>
  </div>
  <div
    className="sc-iwsKbI crUSs"
  >
    <div
      className="sc-gzVnrw hIFWtz"
    >
      <div
        className="sc-htoDjs kvMyWV"
      >
        <img
          alt="loading"
          src="imgLoader.png"
        />
      </div>
      <div
        className="sc-bZQynM gJELVL"
      >
        <img
          alt="Add to favorites"
          className="sc-dnqmqq iKkpJI"
          onClick={[Function]}
          src="starActive.svg"
        />
      </div>
    </div>
    <div
      className="sc-gzVnrw hIFWtz"
    >
      <div
        className="sc-htoDjs kvMyWV"
      >
        <img
          alt="loading"
          src="imgLoader.png"
        />
      </div>
      <div
        className="sc-bZQynM gJELVL"
      >
        <img
          alt="Add to favorites"
          className="sc-dnqmqq iKkpJI"
          onClick={[Function]}
          src="starActive.svg"
        />
      </div>
    </div>
  </div>
</div>
`;
